server/src/test/java/org/elasticsearch/index/store/IndexStoreTests.java

```diff
@@ -75,7 +79,7 @@ private void doTestStoreDirectory(Index index, Path tempDir, String typeSettingV
                     break;
                 case FS:
                     if (Constants.JRE_IS_64BIT && MMapDirectory.UNMAP_SUPPORTED) {
-                        assertTrue(directory.toString(), directory instanceof MMapDirectory);
+                        assertTrue(directory.toString(), directory instanceof FileSwitchDirectory);
```

Comment:
in the two above cases, maybe also check that directory.getPrimaryDir() is a MMapDirectory?

Reply:
I added an additional assertion.